src/java-interop/java-interop-gc-bridge-mono.cc

```diff
@@ -1272,7 +1191,7 @@ gc_cross_references (int num_sccs, MonoGCBridgeSCC **sccs, int num_xrefs, MonoGC
 }
 
 int
-java_interop_gc_bridge_register_hooks (JavaInteropGCBridge *bridge, int weak_ref_kind)
+java_interop_gc_bridge_register_hooks (JavaInteropGCBridge *bridge, [[maybe_unused]] int weak_ref_kind)
```

Comment:
I haven't tried building this, but I'm kinda shocked at the idea that this would build and/or run, considering that you updated the header file to remove the second parameter.

So of course I need to play around…

```cxx
extern "C" {
    void foo();
}

void foo([[maybe_unused]] int x) {
}

int main() {
    foo(0);
}
```

It compiles without error:

```
% clang++ mu.cc -std=c++17
```

…but the symbols, they're mangled!

```
% nm a.out   
0000000100003f80 T __Z3fooi
0000000100000000 T __mh_execute_header
0000000100003f90 T _main
```

There is no `foo` symbol, but there *is* `__Z3fooi`, the C++ mangled version of `foo(int)`.

My *guess* -- again, without compiling this PR -- is that `src/java-interop` will *compile*, but the resulting native library will *not* contain a C callable `java_interop_gc_bridge_register_hooks`, but instead a C++ mangled `__Z37java_interop_gc_bridge_register_hooks…` symbol, and things will blow up at runtime because the P/Invoke for `java_interop_gc_bridge_register_hooks()` won't find that symbol.